src/main/kotlin/no/skatteetaten/aurora/fergus/service/StorageGridService.kt

```diff
@@ -1,36 +1,338 @@
+        storageGridGroupsApi: GroupsApi
+    ): String? {
+        val bucketStatement = PolicyS3Statement()
+            .effect(PolicyS3Statement.EffectEnum.ALLOW)
+            .addActionItem("s3:ListBucket")
+            .addActionItem("s3:GetBucketLocation")
+            .addResourceItem("arn:aws:s3:::$bucketName/*")
+        val objectActionStatement = createS3ObjectActionStatement(bucketName, path, access)
+
+        val postGroupRequest = PostGroupRequest()
+            .displayName(groupName)
+            .policies(
+                Policies().s3(
+                    PolicyS3()
+                        .id(groupName)
+                        .addStatementItem(bucketStatement)
+                        .addStatementItem(objectActionStatement)
+                )
+            )
+        val groupCreateResponse = storageGridGroupsApi
+            .orgGroupsPost(postGroupRequest)
+            .awaitSingle()
+        if (groupCreateResponse.status === GetPatchPostPutGroupResponse.StatusEnum.ERROR) {
+            throw ResponseStatusException(
+                HttpStatus.INTERNAL_SERVER_ERROR,
+                "The Storagegrid groups api returned an error on orgGroupsPost"
+            )
+        }
+
+        return groupCreateResponse.data.id
+    }
+
+    private fun createGroupName(
+        bucketName: String,
+        path: String,
+        access: List<Access>
+    ): String {
+        var groupNamePostfix = ""
+        if (access.size > 0) {
+            access.forEach { groupNamePostfix += it.name.take(1) }
+        } else groupNamePostfix = "RWD"
+
+        val groupName = "$bucketName-$path-$groupNamePostfix"
+        return groupName
+    }
+
+    private fun createS3ObjectActionStatement(
+        bucketName: String,
+        path: String,
+        access: List<Access>
+    ): PolicyS3Statement {
+        var objectActionStatement = PolicyS3Statement()
+            .effect(PolicyS3Statement.EffectEnum.ALLOW)
+            .addResourceItem("arn:aws:s3:::$bucketName/$path/*")
+        if (access.size <= 0) {
+            objectActionStatement
+                .addActionItem("s3:PutObject")
+                .addActionItem("s3:GetObject")
+                .addActionItem("s3:DeleteObject")
+            return objectActionStatement
+        }
+
+        access.forEach {
+            when (it) {
+                Access.READ -> objectActionStatement.addActionItem("s3:GetObject")
+                Access.WRITE -> objectActionStatement.addActionItem("s3:PutObject")
+                Access.DELETE -> objectActionStatement.addActionItem("s3:DeleteObject")
+            }
+        }
+        return objectActionStatement
+    }
+
+    override suspend fun provideUser(
+        userName: String,
+        groupId: UUID,
+        token: String
+    ): UUID {
+        val storageGridUsersApi = storageGridApiFactory.storageGridUsersApi(token)
+        // Get list of buckets for tenant
+        val listUsersResponse = storageGridUsersApi
```

Comment:
Må vi laste ned alle brukerne her? Kunne vi ikke like gjerne be om den aktuelle brukeren?

Reply:
> Må vi laste ned alle brukerne her? Kunne vi ikke like gjerne be om den aktuelle brukeren?

Dette er bare et spørsmål, jeg mener ikke at vi nødvendigvis trenger å endre på noe.

Reply:
Ved nærmere studie av APIet, så kan de se ut til at man kan bruke storageGridUsersApi.orgUsersUserShortNameGet() her.

Reply:
fixed